activesupport/lib/active_support/core_ext/range/compare_range.rb

```diff
@@ -15,6 +15,7 @@ module CompareWithRange
     # The given range must be fully bounded, with both start and end.
     def ===(value)
       if value.is_a?(::Range)
+        return false if value.first && value.last && value.first > value.last
```

Comment:
We should use `begin` and `end` instead of `first` and `last`, for compatibility with beginless and endless ranges:

```suggestion
        return false if value.begin && value.end && value.begin > value.end
```

```ruby
irb(main):001:0> (..1).first
RangeError (cannot get the first element of beginless range)
irb(main):002:0> (..1).begin
=> nil
irb(main):003:0> (1..).last
RangeError (cannot get the last element of endless range)
irb(main):004:0> (1..).end
=> nil
```

Reply:
Updated, but note that the methods will fail later on in these methods for endless ranges since the line

```ruby
value_max = !exclude_end? && value.exclude_end? ? value.max : value.last
```

will fail for endless ranges. Perhaps rather than rebuilding this logic to support beginless/endless ranges, we should just call `super` for `RUBY_VERSION >= '2.6'`?

Reply:
That will only work for `cover?`, since `include?` and `===` don't support being given a range even on Ruby 2.6.

You're right that the existing beginless/endless range support is incomplete - good catch! I'd like to keep this patch focused on backwards ranges, but feel free to follow up with a fix for that other problem too, if you're interested.

Reply:
Good point on the other two methods. I'll look at addressing the other beginless/endless issues in a separate PR.

Anything else needed from me on this PR?